apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx

```diff
@@ -139,6 +139,87 @@ const SwapForm = () => {
     }, idleTime);
   }, [idleState]);
 
+  const isAccount = (value: unknown): value is Account =>
+    Boolean(value && typeof value === "object" && "type" in value && value.type === "Account");
+
+  const getTokenAddress = (account: AccountLike | undefined) => {
+    if (isTokenAccount(account)) {
+      return account.token.contractAddress;
+    }
+    if (isAccount(account)) {
+      // Default to native currency
+      return "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee";
+    }
+  };
+
+  const getAccountWalletAddress = (swapTransaction: SwapTransactionType) => {
+    if (swapTransaction.parentAccount) {
+      return swapTransaction.parentAccount.freshAddress;
+    }
+    if (isAccount(swapTransaction.account)) {
+      return swapTransaction.account.freshAddress;
+    }
+  };
+
+  const getSwapWebAppPropsDex = useCallback(() => {
+    const { swap } = swapTransaction;
+    const { to, from } = swap;
+    const transaction = swapTransaction.transaction;
+    const { account: fromAccount, parentAccount: fromParentAccount } = from;
+    const { account: toAccount, parentAccount: toParentAccount } = to;
+    const { feesStrategy } = transaction || {};
+    const { rate } = exchangeRate || {};
+
+    const fromAccountId =
+      fromAccount && accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id;
+    const toAccountId = toAccount && accountToWalletAPIAccount(toAccount, toParentAccount)?.id;
+    const fromAmount =
+      fromAccount &&
+      convertToNonAtomicUnit({
+        amount: transaction?.amount,
+        account: fromAccount,
+      });
+    const toTokenAddress = getTokenAddress(toAccount);
+    const fromTokenAddress = getTokenAddress(fromAccount);
+    const walletAddress = getAccountWalletAddress(swapTransaction);
+
+    const customFeeConfig = transaction && getCustomFeesPerFamily(transaction);
+    // The Swap web app will automatically recreate the transaction with "default" fees.
+    // However, if you wish to use a different fee type, you will need to set it as custom.
+    const isCustomFee =
+      feesStrategy === "slow" || feesStrategy === "fast" || feesStrategy === "custom";
+    return {
+      provider,
+      providerType: "DEX",
+      fromTokenAddress,
+      fromCurrencyId: from?.currency?.id,
+      fromTokenId: from?.account?.id,
+      toCurrencyId: to?.currency?.id,
+      toTokenId: to?.account?.id,
+      fromAccountId,
+      fromParentAccountId: fromParentAccount
+        ? accountToWalletAPIAccount(fromParentAccount)?.id
+        : undefined,
+      toAccountId,
+      toTokenAddress,
+      walletAddress,
+      fromAmountWei: transaction?.amount.toString(),
+      fromAmount: fromAmount?.toString(),
+      rate: rate?.toString(),
+      feeStrategy: (isCustomFee ? "custom" : "medium")?.toUpperCase(),
+      customFeeConfig: customFeeConfig ? JSON.stringify(customFeeConfig) : undefined,
+      pageState,
+      cacheKey: v4(),
+    };
```

Comment:
I think we can use a unique object for DEX and CEX. The most of this logic are shared by the 2 logic, do not forget that we swap token in CEX too.
You can get the most of this info from wallet-api.

```suggestion
    return {
      provider, // shared
      providerType: "DEX", // is really needed?
      fromTokenAddress, // not need
      fromCurrencyId: from?.currency?.id, // not need
      fromTokenId: from?.account?.id, // not need
      toCurrencyId: to?.currency?.id, // not need
      toTokenId: to?.account?.id, //not need
      fromAccountId, //shared
      fromParentAccountId: fromParentAccount
        ? accountToWalletAPIAccount(fromParentAccount)?.id
        : undefined, // not need
      toAccountId, // shared
      toTokenAddress, // not need
      walletAddress, // not sure about this. I guess you can get the info from wallet api or mock the specific adreess in swap-web-app instead here.
      fromAmountWei: transaction?.amount.toString(), // Not sure what is this...
      fromAmount: fromAmount?.toString(), //shared
      rate: rate?.toString(), //shared
      feeStrategy: (isCustomFee ? "custom" : "medium")?.toUpperCase(), //shared
      customFeeConfig: customFeeConfig ? JSON.stringify(customFeeConfig) : undefined, // shared
      pageState, //shared or not need
      cacheKey: v4(), //shared
    };
```

Reply:
I want to pass this information as they are easy to compute when you have your account object. 
right now on the live app we do not have this account object, once we have it we should remove those variables I agree but I tried wallet-api and I could not get all those ids. I added an exemple JSON to show the different values and what they are used for. You see usage here https://github.com/LedgerHQ/swap-live-app/tree/feat/1inch

`{
    "fromAccountId": "80ef239d-11cc-5526-8437-24712215e7e6",  // CEX
    "fromParentAccountId": "87c586d4-a4ee-57a4-89b0-8d582242839e", // required for tokenApproval, can't get it with wallet-API ...
    "fromTokenAddress": "0xCC42724C6683B7E57334c4E856f4c9965ED682bD", // required 1inch API (ledger backend still does not support swaps) Removing for the PR
    "fromAmount": "30",
    "fromAmountWei": "30000000000000000000", // required ledger backend, can't compute it without magnetude
    "fromCurrencyId": "bsc/bep20/matic_token", // required ledger backend for swap/check allowance
    "fromTokenId": "js:2:bsc:0xe197a5A43521Fc8f0842E63C232Dc2d616A7A308:+bsc%2Fbep20%2Fmatic~!underscore!~token", // required to save swap to history extracted from account object not available in live-app right now
    "walletAddress": "0xe197a5A43521Fc8f0842E63C232Dc2d616A7A308", // required for allowance API Ledger
    "toTokenAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee" // // required 1inch API (ledger backend still does not support swaps) Removing for the PR
}`